### ISSUE TITLE
OrRule over suppression (-(a) | b) is undefined behavior

### ISSUE THREAD

**Issue by vonDonnerstein** [vonDonnerstein/StringParserPEG.jl#6]
Currently this is undefined behavior. Either resolve it or document it properly as undefined behavior...

@Hugo-Trentesaux brought this up originally here: #3 

A condensed non-working example which fails critically:
```
parse(Grammar("""
       start                 => (*(' ') & *(node))
       node                 => ('<' & ( -('#') | ';'))
       """),"<#")
```

And an even more condensed example which fails non-critically:
```
parse(Grammar("start => (-('#') | ';')"),"#")
```

**Comment by vonDonnerstein** [comment]
Maybe it's as simple as this. @Hugo-Trentesaux Can you let me know if this works as expected for you?

**Comment by Hugo-Trentesaux** [comment]
It works with the example you give, but with the example I provided in #3, it runs forever using 100% of one CPU. This [halting problem](https://en.wikipedia.org/wiki/Halting_problem) is an other issue I faced sometimes without trying to understand it. Is it an expected behavior? (I mean, is there a way to detect if a parse operation will lead to an infinite loop in this case)

**Comment by vonDonnerstein** [comment]
That should be an unrelated problem that I just opened another issue for. I will have to think about it a bit longer. For now, can you just eliminate one of your ZeroOrMore rules? For example in your minimal example it seems `text` should really be OneOrMore rather than ZeroOrMore:
```
parse(Grammar("""
       start                 => (br & (*( node | text)))
       node                 => ('<' & (htag|semicol) & '>')
       htag                => -('#')
       semicol                => ';'
       text                 => r([^<>]+)r
       br                => r(\\n*)r
       """),"<#>")
```
(Note: text => r([^<>]**+**)r)

**Comment by Hugo-Trentesaux** [comment]
It was an example specially built to show the problem addressed in this issue, so I guess the problem is solved and we can move to #7

**Comment by vonDonnerstein** [comment]
Fantastic, then I'll close this :)
